src/main/java/hasebo/scrumpoker/controller/VotingWsController.java

```diff
@@ -0,0 +1,151 @@
+import org.springframework.messaging.handler.annotation.SendTo;
+import org.springframework.messaging.simp.SimpMessagingTemplate;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+
+import java.util.*;
+
+@Controller
+@AllArgsConstructor
+public class VotingWsController {
+
+    private final RoomService roomService;
+    private final MemberService memberService;
+    private final VoteRepository voteRepository;
+    private final VotingRepository votingRepository;
+    private final RoomRepository roomRepository;
+    private final CardRepository cardRepository;
+    private final SimpMessagingTemplate simpMessagingTemplate;
+
+    @MessageMapping("/hello")
+    @SendTo("/topic/votes")
+    public String voting(String vote) throws Exception {
+        return vote;
+    }
+
+    @GetMapping("/votingws/{code}")
+    public String voting(@PathVariable("code") String code,
+                         HttpSession httpSession,
+                         Model model) {
+        Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+        model.addAttribute("member", auth.getName());
+        Room votingRoom = roomService.getRoomInfoByCode(code);
+        ArrayList<Card> roomCards = new ArrayList<>((Collection) votingRoom.getCards());
+        model.addAttribute("room", votingRoom);
+        model.addAttribute("roomCards", roomCards);
+        Card card = (Card) httpSession.getAttribute("selectedCard");
+        model.addAttribute("selectedCard", card);
+
+        String username = auth.getName();
+        Member currentMember = memberService.getMemberByName(username);
+        model.addAttribute("voter", currentMember.getName());
+//        if (!room.getVoters().contains(currentMember)) {
+//            room.getVoters().add(currentMember);
+//        }
+//        roomRepository.save(room);
+//        List<Member> voters = room.getVoters();
+//        model.addAttribute("voters", voters);
+
+        Optional<Voting> optionalVoting = votingRepository.findById(1L);
+        Optional<Vote> existingVoteOptional = voteRepository.findByVoterAndVotingAndRoom(currentMember, optionalVoting.get(), votingRoom);
+        if(!existingVoteOptional.isPresent()){
+            Vote vote = new Vote(currentMember, votingRoom, card);
+            optionalVoting.ifPresent(vote::setVoting);
+            voteRepository.save(vote);
+        } else {
+            Vote existingVote = existingVoteOptional.get();
+            Card cardVote = existingVote.getVote();
+            if(cardVote!=null){
```

Comment:
cały ten if można zastąpić
```java
model.addAttribute("selectedCard", cardVote);
```

bo to jest tylko pobieranie karty z bazy, pomimo tego że ją masz